msal/src/request/java/com/microsoft/identity/client/InteractiveRequest.java

```diff
@@ -127,6 +140,73 @@ synchronized void preTokenRequest() throws MsalUserCancelException, MsalClientEx
         processAuthorizationResult(sAuthorizationResult);
     }
 
+    /**
+     * Create the {@Link MicrosoftStsAuthorizationRequest} from authentication parameters.
+     *
+     * @return MicrosoftStsAuthorizationRequest
+     * @throws MsalClientException
+     */
+    private MicrosoftStsAuthorizationRequest createAuthRequest() throws MsalClientException {
```

Comment:
This is coupled awfully tight. Why not have `InteractiveRequest` accept an `AuthorizationRequest` (superclass) as a parameter?

Inside `InteractiveRequest`, `createAuthRequest()` is behaving like a Factory...